docs/paper/dev/api/recipes.mdx

```diff
@@ -0,0 +1,78 @@
+---
+slug: /dev/recipes
+description: How to create and manage recipes.
+---
+
+# Recipes
+
+Recipes are a way to define a way to craft a particular item. They are defined by a plugin.
+
+## <Javadoc name={"org.bukkit.inventory.ShapedRecipe"}>`ShapedRecipe`</Javadoc>
+
+A shaped recipe is a recipe that requires a specific pattern of items in the crafting grid to craft an item.
+These are created using a pattern string and a map of characters to items. The pattern strings are 3,
+3 character strings that represent the rows of the crafting grid. They can be created as follows:
+
+```java
+public class TestPlugin extends JavaPlugin {
+    @Override
+    public void onEnable() {
+        NamespacedKey key = new NamespacedKey(this, "WarriorSword");
+        ItemStack item = new ItemStack(Material.DIAMOND_SWORD);
+
+        ShapedRecipe recipe = new ShapedRecipe(key, item);
+        recipe.shape(" A ", "AAA", " B ");
+        recipe.setIngredient('A', Material.DIAMOND);
+        recipe.setIngredient('B', Material.STICK);
+
+        getServer().addRecipe(recipe);
+    }
+}
+```
+
+This recipe would require a diamond sword to be crafted with a diamond in the top row, a stick in
+the middle row, and a diamond in the bottom row. The diamond sword would be in the middle column of
+the bottom row. This would look like this in the crafting grid:
+
+```
+ A
+AAA
+ B
+```
+
+:::info
+
+You do not need to register the recipe within your plugin's `onEnable` method. You can register it at any time.
```

Comment:
This should mention that if you register it after onEnable, and with players online that you need to resend the recipes or use the boolean addRecipe method https://jd.papermc.io/paper/1.20.6/org/bukkit/Bukkit.html#addRecipe(org.bukkit.inventory.Recipe,boolean)